src/toil/job.py

```diff
@@ -1698,3 +1783,59 @@ def _resolve(cls, jobStoreString, jobStoreFileID):
             # corrupted
             value = cPickle.load(fileHandle)
             return value
+
+
+class PromisedRequirement(object):
+    def __init__(self, *args):
+        """
+        Class for manipulating Promise instances.
+        (see toil.job.Promise)
+
+        :param args[0]: function to act on instance args
+                        default: lambda x: x
+        :param int|Promise args[1:]: list of args, ordered with respect to function parameters
+        """
+        if hasattr(args[0], '__call__'):
+            func = args[0]
+            args = args[1:]
+        else:
+            func = lambda x: x
```

Comment:
What happens if I call PromisedRequirement(1,2)? Is that a legal way to call the constructor?